src/cocoa/toga_cocoa/widgets/canvas.py

```diff
@@ -0,0 +1,188 @@
+from toga_cocoa.libs import (
+    core_graphics,
+    core_text,
+    CGPathDrawingMode,
+    CGRectMake,
+    CGRect,
+    CGSize,
+    CGPoint,
+    CFRange,
+    kCGPathStroke,
+    kCGPathEOFill,
+    kCGPathFill,
+    NSGraphicsContext,
+    NSAffineTransform,
+    NSView,
+    NSRect,
+    objc_method,
+)
+
+from .base import Widget
+
+
+class TogaCanvas(NSView):
+    @objc_method
+    def drawRect_(self, rect: NSRect) -> None:
+        context = NSGraphicsContext.currentContext.graphicsPort()
+
+        # Flip the coordinate system back to normal (unflipped)
```

Comment:
Can this be done at the view level by defining `isFlipped:` (same as is done for TogaBox)?

Reply:
I added an `if self.isFlipped:` check, I hope that is what you mean.

Reply:
No - I meant that if you have an "isFlipped" method on the object, that can set the flip direction permanently on the widget, so you don't need to flip coordinates on every draw. TogaBox has an example of this method; unless I'm missing something, the same should work here.

Reply:
Got it, updated it to add the new method, thanks!